assets/src/application/ui/input/EspressoButton/index.tsx

```diff
@@ -9,46 +9,69 @@ import { EspressoIcon, isEspressoIcon } from '../../display';
  * Button wrapper for adding styles
  */
 const EspressoButton: React.FC<EspressoButtonProps> = ({
-	buttonText,
 	icon,
 	onClick,
-	size = EspressoButtonSize.DEFAULT,
-	btnType = EspressoButtonType.DEFAULT,
+	tooltip,
 	htmlClass,
+	buttonText,
+	size = EspressoButtonSize.DEFAULT,
+	buttonType = EspressoButtonType.DEFAULT,
+	tooltipProps = {},
 	...buttonProps
 }) => {
 	let classes = classNames({
 		[htmlClass]: htmlClass,
-		'esprs-btn-accent': btnType === EspressoButtonType.ACCENT,
-		'esprs-btn-default': btnType === EspressoButtonType.DEFAULT,
-		'esprs-btn-primary': btnType === EspressoButtonType.PRIMARY,
-		'esprs-btn-secondary': btnType === EspressoButtonType.SECONDARY,
+		'esprs-button': true,
+		'esprs-btn-accent': buttonType === EspressoButtonType.ACCENT,
+		'esprs-btn-default': buttonType === EspressoButtonType.DEFAULT,
+		'esprs-btn-primary': buttonType === EspressoButtonType.PRIMARY,
+		'esprs-btn-secondary': buttonType === EspressoButtonType.SECONDARY,
 		'esprs-btn-tiny': size === EspressoButtonSize.TINY,
 		'esprs-btn-small': size === EspressoButtonSize.SMALL,
 		'esprs-btn-big': size === EspressoButtonSize.BIG,
 		'esprs-btn-huge': size === EspressoButtonSize.HUGE,
 		'ee-noIcon': !icon,
 	});
-	let renderedIcon = null;
-	if (isEspressoIcon(icon)) {
-		const svgSize = buttonText ? size : 16;
-		renderedIcon = () => <EspressoIcon icon={icon} svgSize={svgSize} />;
-		if (renderedIcon) {
-			classes += buttonText ? 'esprs-button' : ' ant-btn-icon-only';
-			return (
-				<Button {...buttonProps} onClick={onClick} className={classes}>
+	let eeButton: JSX.Element;
+	// check if icon prop is just an icon name (like "calendar") and if not, assume it is JSX
+	if (typeof icon === 'string') {
+		// custom EE icon
+		if (isEspressoIcon(icon)) {
+			const svgSize = buttonText ? size : 16;
+			const renderedIcon = () => <EspressoIcon icon={icon} svgSize={svgSize} />;
+			if (renderedIcon) {
+				classes += buttonText ? '' : ' ant-btn-icon-only';
```

Comment:
maybe keep this as an object and parse with `classNames` right before applying to button?